src/main/java/cli/screens/CourseEnrollScreen.java

```diff
@@ -0,0 +1,75 @@
+package cli.screens;
+
+import cli.CliApplication;
+import org.example.models.tables.pojos.Course;
+import org.example.models.tables.pojos.Student;
+import org.jooq.DSLContext;
+import storage.PostgresConnectionFactory;
+import student.command.EnrollInCourseCommand;
+import student.query.FindCoursesQuery;
+import student.storage.PostgreSqlStudentRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class CourseEnrollScreen implements Screen{
+    private Student student;
+
+    public CourseEnrollScreen(Student student) {
+        this.student = student;
+    }
+
+    @Override
+    public void show(Scanner in) {
+
+        while (true) {
+            List<String> courseIds = printOpenCourses();
+
+            System.out.print("Course ID: ");
+
+            DSLContext sql = PostgresConnectionFactory.build();
+            var repository = new PostgreSqlStudentRepository(sql);
```

Comment:
Pass this down in the constructor as a dependency instead.

Look up `Inversion of Control` or `Dependency Injection` for reasoning.

Useful explanation: https://stackoverflow.com/questions/3058/what-is-inversion-of-control